File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_android/android/src/main/kotlin/com/theoplayer/flutter/DebugFlagsBridge.kt
package com.theoplayer.flutter

import com.theoplayer.android.api.util.Logger
import com.theoplayer.flutter.pigeon.DebugFlagPigeon
import com.theoplayer.flutter.pigeon.THEOplayerNativeDebugFlagsAPI

class DebugFlagsBridge(
    private val pigeonMessenger: PigeonBinaryMessengerWrapper,
    private val logger: Logger
) : THEOplayerNativeDebugFlagsAPI {

    init {
        THEOplayerNativeDebugFlagsAPI.setUp(pigeonMessenger, this)
    }

    fun dispose() {
        THEOplayerNativeDebugFlagsAPI.setUp(pigeonMessenger, null)
    }

    // MARK: - THEOplayerNativeDebugFlagsAPI

    override fun getAvailableFlags(): List<DebugFlagPigeon> {
        return logger.availableTags().map { tag ->
            DebugFlagPigeon(
                key = tag,
                description = tag,
                defaultValue = false,
                isEnabled = logger.isTagEnabled(tag)
            )
        }
    }

    override fun enableFlag(key: String) {
        logger.enableTags(key)
    }

    override fun disableFlag(key: String) {
        logger.disableTags(key)
    }

    override fun enableAll() {
        logger.enableAllTags()
    }

    override fun disableAll() {
        logger.disableAllTags()
    }

    override fun resetAll() {
        logger.disableAllTags()
    }

    override fun enableFileLogging() {
        // No-op on Android — file logging is iOS-only.
    }
}

File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_ios/ios/theoplayer_ios/Sources/theoplayer_ios/DebugFlagsBridge.swift
//
//  DebugFlagsBridge.swift
//
//  Bridge between Flutter Pigeon API and iOS DebugConfig.
//

import Foundation
import Flutter
@_spi(Debug) import THEOplayerTHEOliveIntegration

class DebugFlagsBridge: THEOplayerNativeDebugFlagsAPI {

    private let pigeonMessenger: PigeonBinaryMessengerWrapper

    init(pigeonMessenger: PigeonBinaryMessengerWrapper) {
        self.pigeonMessenger = pigeonMessenger
        THEOplayerNativeDebugFlagsAPISetup.setUp(binaryMessenger: pigeonMessenger, api: self)
    }

    func dispose() {
        THEOplayerNativeDebugFlagsAPISetup.setUp(binaryMessenger: pigeonMessenger, api: nil)
    }

    // MARK: - THEOplayerNativeDebugFlagsAPI

    func getAvailableFlags() throws -> [DebugFlagPigeon] {
        return DebugConfig.availableFlags.map { flag in
            DebugFlagPigeon(
                key: flag.key,
                description: flag.description,
                defaultValue: flag.defaultValue,
                isEnabled: DebugConfig.isEnabled(flag.key) ?? flag.defaultValue
            )
        }
    }

    func enableFlag(key: String) throws {
        DebugConfig.enable(key)
    }

    func disableFlag(key: String) throws {
        DebugConfig.disable(key)
    }

    func enableAll() throws {
        DebugConfig.enableAll()
    }

    func disableAll() throws {
        DebugConfig.disableAll()
    }

    func resetAll() throws {
        DebugConfig.resetAll()
    }

    func enableFileLogging() throws {
        DebugConfig.enableFileLogging()
    }
}
